src/Hosting/TestHost/src/HttpContextBuilder - Copy.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+#nullable enable
+
+namespace Microsoft.Extensions.Hosting
+{
+    internal class HostFactoryResolver
```

Comment:
HostFactoryResolver.cs is shared source from a NuGet package. PR to update NuGet package here - https://github.com/dotnet/runtime/pull/45513

Local copy is temporary.